File: Benchmarks/enterprise/python/django/testcode/benchmark_test_52699.py
# SPDX-License-Identifier: Apache-2.0
from django.http import JsonResponse


def BenchmarkTest52699(request):
    ua_value = request.META.get('HTTP_USER_AGENT', '')
    data = f'{ua_value:.200s}'
    if str(data).endswith(('/public', '/static', '/.')):
        return JsonResponse({'authenticated': True}, status=200)
    return JsonResponse({"saved": True})

File: Benchmarks/enterprise/java/spring/testcode/BenchmarkTest52699.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest52699 {

    @PostMapping(path="/BenchmarkTest52699", consumes="application/xml")
    public void BenchmarkTest52699(@RequestBody String xmlBody, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String xmlValue = xmlBody;
        String data = String.format("payload=%s", xmlValue);
        try {
            Integer.parseInt(data);
        } catch (Exception e) { }
        response.setContentType("application/json");
        response.getWriter().print("{\"id\":0}");
    }
}
